scan-command/src/main/java/io/ballerina/scan/utils/ScanUtils.java

```diff
@@ -276,6 +301,242 @@ private static void unzipReportResources(InputStream source, File target) throws
+
+        if (Files.exists(cachePath)) {
+            outputStream.println("Loading scan tool configurations from cache...");
+            return loadScanFile(targetDir, cachePath, outputStream);
+        }
+
+        try {
+            FileUtils.copyURLToFile(remoteScanTomlFilePath, cachePath.toFile());
+        } catch (IOException ex) {
+            throw new RuntimeException("Failed to download remote configuration file with exception: "
+                    + ex.getMessage());
+        }
+        outputStream.println("Loading scan tool configurations from " + remoteScanTomlFilePath.toString());
+        return loadScanFile(targetDir, cachePath, outputStream);
+    }
+
+    /**
+     * Returns the {@link ScanTomlFile} representation of the Scan.toml file.
+     *
+     * @param targetDir        the target directory
+     * @param scanTomlFilePath the local Scan.toml file path
+     * @param outputStream     the print stream
+     * @return an in-memory representation of the Scan.toml file
+     */
+    private static ScanTomlFile loadScanFile(Path targetDir, Path scanTomlFilePath, PrintStream outputStream) {
+        Toml scanTomlDocumentContent;
+
+        try {
+            scanTomlDocumentContent = Toml.read(scanTomlFilePath);
+        } catch (IOException ex) {
+            throw new RuntimeException("Failed to read the configuration file with exception: " + ex.getMessage());
+        }
+        ScanTomlFile scanTomlFile = new ScanTomlFile();
+
+        List<Toml> platformTables = scanTomlDocumentContent.getTables(PLATFORM_TABLE);
+        platformTables.forEach(platformTable -> {
+            Map<String, Object> properties = platformTable.toMap();
+            String name = !(properties.get("name") instanceof String) ? null :
+                    properties.remove("name").toString();
+            String path = !(properties.get("path") instanceof String) ? null :
+                    properties.remove("path").toString();
+
+            if (name != null && !name.isEmpty() && path != null) {
+                if (!(new File(path).exists())) {
+                    try {
+                        URL url = new URL(path);
+                        path = loadRemoteJAR(targetDir, name, url, outputStream);
+                    } catch (MalformedURLException ex) {
+                        throw new RuntimeException("Failed to retrieve remote platform file with exception: "
+                                + ex.getMessage());
+                    }
+                }
+
+                if (Files.exists(Path.of(path))) {
+                    ScanTomlFile.Platform platform = new ScanTomlFile.Platform(name, path, properties);
+                    scanTomlFile.setPlatform(platform);
+                }
+            }
+        });
+
+        List<Toml> staticCodeAnalyzerTables = scanTomlDocumentContent.getTables(ANALYZER_TABLE);
+        staticCodeAnalyzerTables.forEach(staticCodeAnalyzer -> {
```

Comment:
Generally, normal for loops may be better since they'll allow continuing early too.